pkg/controller/identity/iamrole/controller.go

```diff
@@ -81,86 +87,101 @@ func (conn *connector) Connect(ctx context.Context, mgd resource.Managed) (manag
 	if err != nil {
 		return nil, errors.Wrap(err, errClient)
 	}
-	return &external{c}, nil
+	return &external{c, conn.client}, nil
 }
 
 type external struct {
 	client iam.RoleClient
+	kube   client.Client
 }
 
 func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
-	cr, ok := mgd.(*v1alpha3.IAMRole)
+	cr, ok := mgd.(*v1beta1.IAMRole)
 	if !ok {
 		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
 	}
 
 	req := e.client.GetRoleRequest(&awsiam.GetRoleInput{
-		RoleName: aws.String(cr.Spec.RoleName),
+		RoleName: aws.String(meta.GetExternalName(cr)),
 	})
 
 	observed, err := req.Send(ctx)
 
-	if iam.IsErrorNotFound(err) {
-		return managed.ExternalObservation{
-			ResourceExists: false,
-		}, nil
+	if err != nil {
+		return managed.ExternalObservation{}, errors.Wrap(resource.Ignore(iam.IsErrorNotFound, err), errGet)
 	}
 
-	if err != nil {
-		return managed.ExternalObservation{}, errors.Wrapf(err, errGet, cr.Spec.RoleName)
+	role := *observed.Role
```

Comment:
@muvaf I think checking for `err` should be enough. If there is no Role, `err` is non-nil.

Reply:
That assumes everything works correct in AWS SDK though. Not a big deal, but as a general rule of thumb, I'd go with the fact that if something is pointer then it should be used after we make sure it has a value. Because dereferencing `nil` brings down the whole manager with all of its controllers by `panic`ing.

Reply:
Looks like this still needs to be checked @sahil-lakhwani :)